### PR TITLE
Rename presto references to Trino, upgrade SPI

### PR DESCRIPTION
**Pull request by alec-heif**
As of Presto 351, the project has been renamed to Trino and plugins need to update to use the latest SPI to remain compatible.

This PR performs the SPI upgrade and renames all presto references to be Trino (with the exception of references which refer to the plugin's name on Github).

I'll leave it to your discretion whether to accept this PR (which will make `presto-querylog` incompatible with pre-351 Trino releases) or whether to reject it (which will make `presto-querylog` incompatible with Trino 351 onward).

### CODE REVIEW

**Comment by alec-heif** [comment]
whoops, looks like I need to change something about Sonar to continue using the `presto-querylog` name

**Comment by alec-heif** [comment]
actually it looks like only the original project author @rchukh has permissions to run said Sonar job on `com.github.rcchuck.presto-querylog`, so I guess this will fail for any PRs from external contributors

**Comment by rchukh** [comment]
@alec-heif Thank you for the PR.

Yes, it makes sense to migrate to Trino (this project was originally migrated to PrestoSQL(now Trino) from PrestoDB anyway).

I would expect Sonarcloud to work... but it does not seem to support analysis from external PR's: https://jira.sonarsource.com/browse/MMF-1371

I'll try to run a few tests and maybe update dependencies that transitively coming from SPI as well, and then merge it.

**Comment by alec-heif** [comment]
thanks! let me know if you have any questions, feel free to modify this PR as you see fit

File: README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 # Overview
 
-Presto QueryLog is a [PrestoSQL](https://prestosql.io/) plugin for logging query events into separate log file.
+Trino QueryLog is a [TrinoSQL](https://trino.io/) plugin for logging query events into separate log file.
```
Comment (findepi) [inline README.md]: The project name is just "Trino" (not "TrinoDB" or "TrinoSQL")

```suggestion
Trino QueryLog is a [Trino (formerly Presto SQL)](https://trino.io/) plugin for logging query events into separate log file.
```

**Comment by rchukh** [comment]
@alec-heif 

I've merged it and updated some other minor things in a different PR (as well as repo name/description update).
Haven't run any end to end tests yet, so will release a 0.4 version after that.

Thank you again for the PR.

**Pull request merged**